packages/dds/tree/src/simple-tree/core/treeNodeKernel.ts

```diff
@@ -335,35 +338,6 @@ export class TreeNodeKernel {
 		return this.#hydrationState.innerNode;
 	}
 
-	/**
-	 * Creates an anchor node and associates it with the given proxy.
-	 * @privateRemarks
-	 * Use `forgetters` to cleanup the anchor allocated by this function once the anchor is no longer needed.
-	 * In practice, this happens when either the anchor node is destroyed, or another anchor to the same node is created by a new flex node.
-	 *
-	 * The FlexTreeNode holds a reference to the same anchor, and has a lifetime at least as long as the simple-tree,
-	 * so this would be unnecessary except for the case of "marinated" nodes, which have an anchor,
-	 * but might not have a FlexTreeNode.
-	 * Handling this case is an optimization assuming that this extra anchor reference is cheaper than eagerly creating FlexTreeNodes.
-	 */
-	public anchorProxy(anchors: AnchorSet, path: UpPath): AnchorNode {
-		assert(!anchorForgetters.has(this.node), 0x91c /* Proxy anchor should not be set twice */);
```

Comment:
the moving of assert 0x7f5 into createHydratedState covers detecting this case now.